File: src/test/java/com/automation/um/pages/components/ToastComponent.java
package com.automation.um.pages.components;

import com.automation.um.pages.BasePage;
import org.openqa.selenium.WebDriver;

public class ToastComponent extends BasePage {
  public ToastComponent(WebDriver driver) {
    super(driver);
  }

  public boolean toastVisible(String variant) {
    return isVisible("toast-" + variant);
  }

  public boolean hasTitle(String variant, String title) {
    if (!toastVisible(variant)) {
      return false;
    }
    return el("toast-" + variant).getText().contains(title);
  }
}

File: playwright/pages/components/ToastComponent.js
import { BasePage } from "../BasePage.js";

export class ToastComponent extends BasePage {
  async toastVisible(variant) {
    return this.isVisible(`toast-${variant}`);
  }

  async hasTitle(variant, title) {
    if (!(await this.toastVisible(variant))) return false;
    const text = await this.testId(`toast-${variant}`).textContent();
    return text?.includes(title) ?? false;
  }
}
